vizro-ai/src/vizro_ai/dashboard/nodes/core_builder/plan.py

```diff
@@ -0,0 +1,321 @@
+                description="Target component to be affected by filter. If none are given "
+                "then target all components on the page that use `column`.",
+            ),
+        ),
+        column=(str, Field(..., description="Column name of DataFrame to filter. ALWAYS REQUIRED.")),
+        __validators__={
+            "validator1": validator("targets", pre=True, each_item=True, allow_reuse=True)(validate_targets),
+            "validator2": validator("column", allow_reuse=True)(validate_column),
+        },
+        __base__=Filter,
+    )
+
+
+class Control(BaseModel):
+    """Control plan model."""
+
+    control_type: control_type
+    control_description: str = Field(
+        ...,
+        description="Description of the control. Include everything that seems to relate to this control."
+        "Be as detailed as possible. Keep the original relevant description AS IS. If this control is used"
+        "to control a specific component, include the relevant component details.",
+    )
+    data_frame: str = Field(
+        ...,
+        description="The name of the dataframe that this component will use. "
+        "If the dataframe is not used, please specify that.",
+    )
+
+    # TODO: there is definitely room for dynamic model creation, e.g. with literals for targets
+    def create(self, model, available_components, df_metadata):
+        """Create the control."""
+        filter_prompt = (
+            f"Create a filter from the following instructions: {self.control_description}. Do not make up "
+            f"things that are optional and DO NOT configure actions, action triggers or action chains."
+            f" If no options are specified, leave them out."
+        )
+        try:
+            _df_schema, _df_sample = (
+                df_metadata[self.data_frame]["df_schema"],
+                df_metadata[self.data_frame]["df_sample"],
+            )
+            _df_cols = list(_df_schema.keys())
+        # when wrong dataframe name is given
+        except KeyError:
+            logger.info(f"Dataframe {self.data_frame} not found in metadata, returning default values.")
+            return None
+
+        try:
+            result_proxy = create_filter_proxy(
+                df_cols=_df_cols, df_sample=_df_sample, available_components=available_components
+            )
+            proxy = _get_model(query=filter_prompt, model=model, result_model=result_proxy, df_metadata=df_metadata)
+            logger.info(
+                f"`Control` proxy: {proxy.dict()}"
+            )  # when wrong column name is given, `AttributeError: 'ValidationError' object has no attribute 'dict'``
+            actual = Filter.parse_obj(
+                proxy.dict(
+                    exclude={"selector": {"id": True, "actions": True, "_add_key": True}, "id": True, "type": True}
+                )
+            )
+            # del model_manager._ModelManager__models[proxy.id]  # TODO: This is very wrong and needs to change
```

Comment:
Are these proxy models meant to have `id` field? Ultimately that is why they are added to the model manager